types/babel__traverse/index.d.ts

```diff
@@ -285,19 +288,21 @@ export class NodePath<T = Node> {
      * transforming ASTs is an antipattern and SHOULD NOT be encouraged. Even if it's
      * easier to use, your transforms will be extremely brittle.
      */
-    replaceWithSourceString(replacement: any): void;
+    replaceWithSourceString(replacement: any): [NodePath];
 
     /** Replace the current node with another. */
-    replaceWith(replacement: Node | NodePath): void;
+    replaceWith<T extends Node>(replacement: T | NodePath<T>): [NodePath<T>];
 
     /**
      * This method takes an array of statements nodes and then explodes it
      * into expressions. This method retains completion records which is
      * extremely important to retain original semantics.
      */
-    replaceExpressionWithStatements(nodes: Node[]): Node;
+    replaceExpressionWithStatements<Nodes extends Node[]>(nodes: Nodes): NodePaths<Nodes>;
 
-    replaceInline(nodes: Node | Node[]): void;
+    replaceInline<T extends Node | Node[]>(
+        nodes: T,
+    ): T extends Node[] ? { [K in keyof T]: NodePath<T[K]> } : [NodePath<T>];
```

Comment:
Can you double check this return type? The mapped type over an array type looks strange to me. Should the true type not be `NodePaths<T>`?

Reply:
Oh, I think I didn't realize that `NodePaths` allows `T` to be a non-array

Reply:
That or maybe tests were failing in some weird way?  There must be some reason...I'll check in a bit

Reply:
okay using `NodePaths` seems to work